Address patch v4 comments

ndev2 opened the pull request:
Hi Cody, I've addressed the comments. But should `.parse_copt` interface also have `const char *dev` instead of `char *dev`, I'm not sure so do give your thoughts on this, thanks.

Commit:
Address patch v4 comments

cjdoucette commented on lib/utils.c:
```diff
@@ -71,15 +71,15 @@ int read_prop(const char *dev, char *prop, long *value)
 	}
 
-	if (result == ULONG_MAX && errno == ERANGE) {
+	if (result == LONG_MAX && errno == ERANGE) {
```
Since we're now converting a string to a signed quantity (`strtol()`), there can be overflow in either direction. So we also need to check whether result is `LONG_MIN`:

    if ((result == LONG_MAX || result == LONG_MIN) && errno == ERANGE) {

cjdoucette commented on tc/q_atm.c:
```diff
@@ -26,7 +26,7 @@
 
 
-static int atm_parse_opt(struct qdisc_util *qu, int argc, char **argv, struct nlmsghdr *n)
+static int atm_parse_opt(struct qdisc_util *qu, int argc, char **argv, struct nlmsghdr *n, const char *dev)
```
Shouldn't this patch only be adding `const` in front of device name variable declarations? This function didn't have a `char *dev` argument in previous patches -- why now?

ndev2 replied:
Not sure why, but the `dev` parameter was missing in the function definition

cjdoucette replied:
I'm noticing that the compiler is only compiling certain files if some options are set. That's why we missed this function. For example, for tc/q_atm.c, this is in the Makefile:

    ifeq ($(TC_CONFIG_ATM),y)
        TCSO += q_atm.so
    endif

Therefore, we should check to make sure we have all of these function headers altered by searching through the code. One way is to look for any files that try to assign a function to the `parse_qopt` function pointer member of `struct qdisc_util`:

    $ grep -lr ".parse_qopt"

Same with with the class options:

    $ grep -lr ".parse_copt"

cjdoucette commented on tc/q_cbs.c:
```diff
@@ -34,7 +34,7 @@ static void explain1(const char *arg, const char *val)
 }
 
-static int cbs_parse_opt(struct qdisc_util *qu, int argc, char **argv, struct nlmsghdr *n)
+static int cbs_parse_opt(struct qdisc_util *qu, int argc, char **argv, struct nlmsghdr *n, const char *dev)
```
Same thing for this function header.

ndev2 replied:
Yes, same thing, the `dev` parameter was missing, sorry

ndev2 replied:
But good thing, we found it before it got applied

cjdoucette commented on tc/tc_util.h:
```diff
@@ -24,7 +24,7 @@ struct qdisc_util {
 	const char *id;
 	int (*parse_qopt)(struct qdisc_util *qu, int argc,
-			  char **argv, struct nlmsghdr *n, char *dev);
+			  char **argv, struct nlmsghdr *n, const char *dev);
```
Yes, let's also add a `const` in the `parse_copt()` signature. The reviewer didn't specify this, but the same logic applies.

ndev2 commented:
Hi Cody, I've updated the code, hope its good, do let me know, thanks.

cjdoucette commented on lib/utils.c:
```diff
@@ -71,15 +71,15 @@ int read_prop(const char *dev, char *prop, long *value)
 
-	if (result == ULONG_MAX && errno == ERANGE) {
+	if ((result == LONG_MAX || result == LONG_MIN) && errno == ERANGE) {
 		fprintf(stderr, "strtoul %s: %s", fname, strerror(errno));
```
This print should now contain "strtol" since we changed the function we're using.

ndev2 commented:
Hey Cody, I've updated the code, hope its good. Thanks for the tips, will keep them in mind, I cross checked where .parse_copt headers are defined and where we've changed these headers, and now we do have all the headers covered with our changes. Thanks. Also, thanks for your patience. Do let me know if the patch is good to be sent out, and I'll do that.

cjdoucette commented:
Looks good, go right ahead!
